lib/matplotlib/tests/test_axes.py

```diff
@@ -4603,6 +4603,64 @@ def test_hist_stacked_bar():
     ax.legend(loc='upper right', bbox_to_anchor=(1.0, 1.0), ncols=1)
 
 
+@pytest.mark.parametrize('kwargs', ({'facecolor': ["b", "g", "r"]},
+                                    {'edgecolor': ["b", "g", "r"]},
+                                    {'hatch': ["/", "\\", "."]},
+                                    {'linestyle': ["-", "--", ":"]},
+                                    {'linewidth': [1, 1.5, 2]},
+                                    {'color': ["b", "g", "r"]}))
+@check_figures_equal(extensions=["png"])
+def test_hist_vectorized_params(fig_test, fig_ref, kwargs):
+    np.random.seed(19680801)
+    x = [np.random.randn(n) for n in [2000, 5000, 10000]]
```

Comment:
```suggestion
    xs = [np.random.randn(n) for n in [20, 50, 100]]
```
Slight preference to use `xs` for the collection of datasets and then `x` for a single dataset, rather than `x` for the collection and `xi` for a dataset. - `xi` is very often used for a single number.

Also let's reduce the dataset size somewhat to not not spend any relevant time on numerics. - Yes creating the figure and axes is currently on the same order as the numeric calculations. But if we can speed up that at some point, we want the test to become faster as well.